### PR TITLE
fix: Sourcify worker duplicate job casue performance problem of pg (issue-1156)

### PR DESCRIPTION
**Pull request by Naupio**
## What problem does this PR solve?
link issue: #1156 
fix SmartContractRegister produce duplicate Sourcify-worker for registering which blocking job queue 
## Check List
- check  smart contract register work automatically
- check sourcify-worker is working
#### Test
- unit test
#### Task
 - none

### CODE REVIEW

File: lib/godwoken_indexer/worker/smart_contract_register.ex [hunk 1]
```diff
@@ -1,5 +1,5 @@
-defmodule GodwokenExplorer.Graphql.Workers.SmartContractRegister do
-  alias GodwokenExplorer.Graphql.Workers.Sourcify, as: ObanSourcify
+defmodule GodwokenIndexer.Workers.SmartContractRegister do
```
Comment (zmcNotafraid) [inline lib/godwoken_indexer/worker/smart_contract_register.ex:1]:
```suggestion
defmodule GodwokenIndexer.Worker.SmartContractRegister do
```

File: lib/godwoken_indexer/worker/smart_contract_register.ex [hunk 2]
```diff
@@ -11,11 +11,16 @@ defmodule GodwokenExplorer.Graphql.Workers.SmartContractRegister do
   def perform(%Oban.Job{}) do
     last_100_unregistered_polyjuice_contracts = get_unregistered_eth_addresses() |> Repo.all()
 
-    Enum.each(last_100_unregistered_polyjuice_contracts, fn account ->
-      account
-      |> ObanSourcify.new()
-      |> Oban.insert()
+    # oban open community version, need insert one by one to avoid duplicate job
```
Comment (zmcNotafraid) [inline lib/godwoken_indexer/worker/smart_contract_register.ex:14]: each 不是 one_by_one 吗？

File: lib/godwoken_indexer/worker/sourcify.ex [hunk 3]
```diff
@@ -1,7 +1,7 @@
-defmodule GodwokenExplorer.Graphql.Workers.Sourcify do
+defmodule GodwokenIndexer.Workers.Sourcify do
   alias GodwokenExplorer.Graphql.Sourcify
 
-  use Oban.Worker, queue: :default
+  use Oban.Worker, queue: :default, unique: [period: :infinity, states: Oban.Job.states()]
```
Comment (zmcNotafraid) [inline lib/godwoken_indexer/worker/sourcify.ex:4]: 加这个就够了把？

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

File: lib/godwoken_indexer/worker/smart_contract_register.ex (hunk 2 above)
Comment (Naupio) [reply lib/godwoken_indexer/worker/smart_contract_register.ex:14]: each 是的啊, 只是这里有话一下 并发 insert 进取, 增加一下备注说明

File: lib/godwoken_indexer/worker/smart_contract_register.ex (hunk 1 above)
Comment (Naupio) [reply lib/godwoken_indexer/worker/smart_contract_register.ex]: fixed

**Review by zmcNotafraid — APPROVED** [review]

**Review by FrederLu — APPROVED** [review]

**Pull request merged**